K2Bridge/Models/Request/Aggregations/Bucket/Histogram/HistogramAggregation.cs

```diff
@@ -0,0 +1,59 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license.
+// See LICENSE file in the project root for full license information.
+
+namespace K2Bridge.Models.Request.Aggregations
+{
+    using K2Bridge.JsonConverters;
+    using K2Bridge.Visitors;
+    using Newtonsoft.Json;
+
+    /// <summary>
+    /// This multi-bucket aggregation is similar to the normal histogram, but it can only be used with date or date range values.
+    /// </summary>
+    internal class HistogramAggregation : BucketAggregation
+    {
+        /// <summary>
+        /// Gets or sets the field to target.
+        /// </summary>
+        [JsonProperty("field")]
+        public string Field { get; set; }
+
+        /// <summary>
+        /// Gets or sets the interval to use when bucketing documents.
+        /// </summary>
+        [JsonProperty("interval")]
+        public int? Interval { get; set; }
+
+        /// <summary>
+        /// Gets or sets the minimum number of documents that a bucket must contain to be returned in the response.
+        /// The default is 0 meaning that buckets with no documents will be returned.
+        /// </summary>
+        [JsonProperty("min_doc_count")]
+        public int? MinimumDocumentCount { get; set; }
+
+        /// <summary>
+        /// Gets or sets field keyed for the Histogram Aggregation.
+        /// </summary>
+        [JsonProperty("keyed")]
+        public bool? Keyed { get; set; } = false;
+
+        /// <summary>
+        /// Gets or sets field extended_bounds for the Histogram Aggregation.
+        /// </summary>
+        [JsonProperty("extended_bounds")]
+        public Bounds ExtendedBounds { get; set; }
```

Comment:
I added extended bounds to my query, and I don't see any difference in the results. This property seems to be unused

Reply:
Based on the [doc](https://www.elastic.co/guide/en/elasticsearch/reference/current/search-aggregations-bucket-histogram-aggregation.html) I understood that it's used only in the UI side for graph generation

Reply:
where is the part that it says its the client's responsibility?

Reply:
I don't think so, it doesn't work with your code but it does work in the demo

Reply:
any suggestion about how to implement this ?

Reply:
I guess that's not very straightforward.

First of all you need to support empty buckets in range ( so if the range is 50-100, and the data is 75-100, you will need to have 50-75 be empty buckets).

Then you will need to calculate the min and max from kusto.

And then have the between be `between ( iff(kusto_min < extended_bounds_min, kusto_min , extended_bounds_min) ..  iff(kusto_max > extended_bounds_max, kusto_max , extended_bounds_max))`

Probably too much for this PR

Reply:
I am closing this thread, I think its resolved